src/orchard/inspect.clj

```diff
@@ -408,17 +408,31 @@
           inspector
           mappable))
 
+(defn- left-pad
+  "TODO: should extract this into a library."
+  [^String s, final-len]
+  (let [len (.length s)]
+    (loop [s s, len len]
+      (if (< len final-len)
+        (recur (str " " s) (inc len))
+        s))))
+
 (defn render-indexed-values
-  ([inspector obj] (render-indexed-values inspector obj 0))
-  ([inspector obj idx-starts-from]
-   (loop [ins inspector, obj (seq obj), idx idx-starts-from]
-     (if obj
-       (recur (-> ins
-                  (render-indent (str idx) ". ")
-                  (render-value (first obj))
-                  (render-ln))
-              (next obj) (inc idx))
-       ins))))
+  "Render an indexed collection of values. Renders all values in `chunk`, so
+  `chunk` must be finite."
+  ([inspector chunk] (render-indexed-values inspector chunk 0))
+  ([inspector chunk idx-starts-from]
+   (let [n (count chunk)
```

Comment:
May it make sense to use bounded-count, when available, for safety?

An altenative being asserting that the chunk is Counted

Reply:
The current version of this function goes over the whole sequence non-conditionally anyway. Adding `count` doesn't really make it less safe. There is currently only one caller of this function that passes in a finite sequence. I renamed the argument to make this more apparent at least documentation-wise. Don't know if it needs any other guardrails.